identity_credential/src/domain_linkage/domain_linkage_credential_builder.rs

```diff
@@ -139,6 +144,18 @@ mod tests {
       .unwrap_err();
     assert!(matches!(err, Error::DomainLinkageError(_)));
   }
+  #[test]
+  fn test_builder_origin_is_a_url() {
+    let issuer: CoreDID = "did:example:issuer".parse().unwrap();
+    let err: Error = DomainLinkageCredentialBuilder::new()
+      .issuance_date(Timestamp::now_utc())
+      .expiration_date(Timestamp::now_utc())
+      .issuer(issuer)
+      .origin(Url::parse("https://example.com/foo?bar=420#baz").unwrap())
```

Comment:
I would add other test cases for `"https://example.com/?bar=420#baz"` and `"https://example.com/#baz"`